### PR TITLE
executor: remove bce in the AnalyzeColumnsExec

### LINKED ISSUE #49704: remove BCE in the AnalyzeColumnsExec
## Enhancement

We propose to remove the BCE part of the code in AnalyzeColumnsExec. This would simplify the code and potentially improve performance. We believe this can be achieved without negatively impacting the functionality of the AnalyzeColumnsExec.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #49704

Problem Summary:

### What changed and how does it work?

We've found that the current implementation of the AnalyzeColumnsExec includes unnecessary BCE (Bounds Check Elimination). so this pr is to remove them.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

before

```
pkg/executor/analyze_col.go:245:21: Found IsInBounds
pkg/executor/analyze_col.go:251:31: Found IsInBounds
pkg/executor/analyze_col.go:252:25: Found IsInBounds
pkg/executor/analyze_col.go:253:25: Found IsInBounds
pkg/executor/analyze_col.go:259:28: Found IsInBounds
pkg/executor/analyze_col.go:326:16: Found IsSliceInBounds

```

after

```
pkg/executor/analyze_col.go:245:21: Found IsInBounds
pkg/executor/analyze_col.go:251:31: Found IsInBounds
pkg/executor/analyze_col.go:252:36: Found IsInBounds
pkg/executor/analyze_col.go:327:16: Found IsSliceInBounds

```

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by elsa0520 — APPROVED** [review]
LGTM

File: pkg/executor/analyze_col.go [hunk 1]
```diff
@@ -249,15 +249,16 @@ func (e *AnalyzeColumnsExec) buildStats(ranges []*ranger.Range, needExtStats boo
 			topNs = append(topNs, collectors[i].TopN)
 		}
 		for j, s := range collectors[i].Samples {
-			collectors[i].Samples[j].Ordinal = j
-			collectors[i].Samples[j].Value, err = tablecodec.DecodeColumnValue(s.Value.GetBytes(), &col.FieldType, timeZone)
+			samples := collectors[i].Samples[j]
```
Comment (lance6716) [inline pkg/executor/analyze_col.go:252]: can we use `s` from above loop?

**Review by winoros — APPROVED** [review]

File: pkg/executor/analyze_col.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/executor/analyze_col.go]: Yes, I fixed.

**Pull request merged**